### ISSUE TITLE
Move ComNativeDescriptor from WinForms to System.ComponentModel.TypeConverter

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/corefx#36498]
Current implementation: https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/ComNativeDescriptor.cs

See https://github.com/dotnet/corefx/pull/36496 for the current work around.

**Comment by AaronRobinsonMSFT** [comment]
cc @safern @maryamariyan @merriemcgaw

**Comment by safern** [comment]
Doesn't look like something we should block on, @AaronRobinsonMSFT is that the only file we should move down to avoid that workaround? Or does ComNativeDescriptor have more dependencies?

cc: @danmosemsft

**Comment by AaronRobinsonMSFT** [comment]
@safern Everything in the `COM2Interop` directory will need to be brought over. There is a lot of code in there and it is all rather stale.

**Comment by ericstj** [comment]
@AaronRobinsonMSFT can you please elaborate on this?  I see plenty of other things in COM2Interop that depend on WinForms types, so we definitely cannot bring all that down.

Previously on desktop there was a loose binding here via a FullyQualifiedTypeName to System.Windows.Forms: https://referencesource.microsoft.com/#System/compmod/system/componentmodel/TypeDescriptor.cs,4689

Why can't we keep that here?  Are you concerned about scenarios where folks need to use this but aren't targeting the WindowsDesktop shared framework?

**Comment by AaronRobinsonMSFT** [comment]
>Why can't we keep that here? Are you concerned about scenarios where folks need to use this but aren't targeting the WindowsDesktop shared framework?

Yep. This is a really loose coupling that is easy to forget about or even miss. There isn't any reason this can't be kept as is, but it is a very jagged edge that would be better to fix if possible. As mentioned above, the fix is basically what was done in framework (https://github.com/dotnet/corefx/pull/36496). The issue with the framework "mitigation" is that the failure would be silent and so was probably never hit. The referenced PR talks about why the fix in CoreFX was done the way it was and why the .NET Framework coupling would have been a silent failure.

**Comment by ericstj** [comment]
I see internals access to ComNativeDescriptor in WinForms that would complicate pushing that type down, also seeing a non-trivial type closure of this type that would cause more types to be pulled down.  I haven't done a full analysis of that yet but it seems like a considerable amount of work that is risky since it would mean cross-repo collaboration.  Unless there is a customer ask to enable this functionality in non-WindowsDesktop targeting apps then I would reccomend we postpone it for now.

**Comment by AaronRobinsonMSFT** [comment]
cc @JeremyKuhne
